test/e2e/specs/editor/blocks/navigation.spec.js

```diff
@@ -48,5 +60,89 @@ test.describe(
 <!-- /wp:navigation -->`
 			);
 		} );
+
+		test( 'default to my only existing menu', async ( {
+			editor,
+			navBlockUtils,
+		} ) => {
+			const createdMenu = await navBlockUtils.createNavigationMenu( {
+				title: 'Test Menu 1',
+				content:
+					'<!-- wp:navigation-link {"label":"WordPress","type":"custom","url":"http://www.wordpress.org/","kind":"custom","isTopLevelLink":true} /-->',
+			} );
+
+			await editor.insertBlock( { name: 'core/navigation' } );
+
+			// Check the markup of the block is correct.
+			await editor.publishPost();
+			const content = await editor.getEditedPostContent();
+			expect( content ).toBe(
+				`<!-- wp:navigation {"ref":${ createdMenu.id }} /-->`
+			);
+			await editor.page
+				.locator( 'role=button[name="Close panel"i]' )
+				.click();
+
+			//check the block in the canvas.
+			await expect(
+				editor.page.locator(
+					'role=gridcell[name="Custom Link link"] >> a:has-text("WordPress")'
+				)
+			).toBeVisible();
+
+			//check the block in the frontend.
+			await editor.page.goto( '/' );
+			await expect(
+				editor.page.locator(
+					'nav:has-text("WordPress") >> role=link[name="WordPress"]'
```

Comment:
Same here, prefer role selectors over CSS selectors whenever possible.

Reply:
it's the frontend here so it was a bit trickier, but I think what I did is fine?